### PR TITLE
[gh394] Remove the Figaro gem dependency

### LINKED ISSUE #394: Remove the Figaro gem dependency
## Why

In order to unlock #316 (Update Ruby 3.1).
- The Figaro gem did not receive any update since 3 years.
- Rails 7 projects now use default fallback values in the `database.yml` file (in case the ENV variable is not set). See [this example](https://github.com/zakariaf/rails-base-app/blob/main/config/database.yml)
- We already have a `.env` file for other default non-sensitive variables (the `application.yml` is actually confusing)
 
## Who Benefits?

Developers will be less confused about where the ENV default should go. Developers with the future possibility of using the next Ruby versions.

### PR DESCRIPTION
**Pull request by malparty**
- close #394

## What happened 👀

Remove the figaro dependency in favor of dotenv gem and a .env.example file

- [x] Update the needed files
- [x] Update the tests (review in progress)
- [x] Test the changes with a deployed project 🤖 

## Insight 📝

`n/a`

## Proof Of Work 📹

Local project generated from this branch: 

image

| Test can run | Project can run |
|---|---|
|image|image|

### CODE REVIEW

**Comment by malparty** [comment]
@malparty you should try that:

image

File: config/database.yml.tt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+test:
+  <<: *default
+  host:     <%= ENV.fetch('DB_HOST') { 'localhost' } %>
+  port:     <%= ENV.fetch('DB_PORT') { 5432 } %>
+  username: <%= ENV.fetch('DB_USERNAME') { 'postgres' } %>
+  password: <%= ENV.fetch('DB_PASSWORD') { 'postgres' } %>
+  database: <%= ENV.fetch('DB_NAME') { APP_NAME } %>_test
+
+production:
+  url: <%= ENV.fetch('DATABASE_URL') { '' } %>
```
Comment (caherrerapa) [inline config/database.yml.tt:23]: this will only work for heroku afaik

Comment (malparty) [reply config/database.yml.tt:23]: Thanks :) 

Andy found a part of the solution in #411 — but the implementation still require some improvements 💭

**Comment by malparty** [comment]
I will close this PR as with the recent changes, it will be wiser to start it from scratch again 👍 
I will add this PR as reference in the associated issue.

**Pull request closed without merging**